src/common.rs

```diff
@@ -36,25 +41,61 @@ impl fmt::Display for FileEntry {
 }
 
 impl FileEntry {
-    /// Converts `PathBuf` to `FileEntry`
-    pub fn from(path: PathBuf) -> Option<Self> {
-        let file_name = match path.file_name() {
+    /// Converts `PathBuf` to tuple of `(FileEntry, Vec<String>)`
+    pub fn from(path: PathBuf) -> Option<(Self, Vec<String>)> {
+        let file_name = match Self::get_file_name(&path) {
             Some(s) => s,
             None => return None,
         };
 
-        let file_name = match file_name.to_str() {
-            Some(s) => s.to_string(),
+        let content = match Self::get_content_by_path(&path) {
+            Some(s) => s,
             None => return None,
         };
 
-        Some(FileEntry { path, file_name })
+        let mut lines: Vec<String> = content.lines().map(|str| str.to_string()).collect();
+
+        // You must add a line, because [`Lines`] does not return the last empty row (excludes LF)
+        if content.ends_with(LF) {
+            lines.push(LF.to_string());
+        }
+
+        Some((
+            FileEntry {
+                path,
+                file_name,
+                total_lines: lines.len(),
+            },
+            lines,
+        ))
     }
 
     /// Checks a file name with the `.env` pattern
-    pub fn is_env_file(&self) -> bool {
+    pub fn is_env_file(path: &PathBuf) -> bool {
         let pattern = ".env";
-        self.file_name.starts_with(pattern) || self.file_name.ends_with(pattern)
+        Self::get_file_name(path)
+            .filter(|file_name| file_name.starts_with(pattern) || file_name.ends_with(pattern))
+            .is_some()
+    }
+
+    fn get_file_name(path: &PathBuf) -> Option<String> {
+        path.file_name()
+            .map(|file_name| file_name.to_str())
+            .filter(|s| s.is_some())
+            .map(|s| s.unwrap().to_string())
```

Comment:
> Please don't use the `unwrap` method, because it might panic.

Yes, that’s why I added `.filter(|s| s.is_some())` before `map()`, but really, it’s probably worth replacing with something more readable.
Thank you!

Reply:
Replaced by a more obvious and safe `unwrap_or(None)` instead `filter().map()`